src/components/skeletons/SkeletonProductGrid/SkeletonProductGrid.tsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+import { ITEMS_PER_PAGE } from 'src/constants'
+
+import SkeletonProductCard from '../SkeletonProductCard'
+
+import 'src/components/product/ProductGrid/product-grid.scss'
+
+interface Props {
+  loading?: boolean
+  children?: JSX.Element
+}
+
+function SkeletonProductGrid({ children, loading = true }: Props) {
+  return loading ? (
+    <ul className="product-grid">
+      {Array.from({ length: ITEMS_PER_PAGE }, (_, index) => (
```

Comment:
That's kinda tricky, why not create and use an iterator?

Reply:
Also, would be better if it receives a property called quantityItens

Reply:
About the Array.from it's ok for me.  About the quantity items, I try to explain better.

```tsx
const ProductShelf = ({ ..., pageSize }) => {
   <ProductGridSkeleton slots={pageSize}> // I think slots is better than quantityItems, idk, feel free to choose a better name
     {/* Children */}
   </ProductSkeletonGrid>
}
```
The ProductShelf component has the responsibility to know how many items he will render, right? This value comes from the SearchProvider (ITEMS_PER_PAGE)
As far as I understand, the SkeletonProductGrid, should render `a grid with slots`, how many? Idk, doesn't matter.
So, knowing this,
```tsx
const ProductGridSkeleton = ({ slots }) => {
  return <ul>{Array.from({ length: slots }, (_, index) => (<ProductCardSkeleton />))}
}
```

Does it make sense? Can go further doing this:

```tsx
const ProductShelf = ({pageSize, products, ....}) => {
  const loading = products?.length === 0
  
  return loading ? (<ProductGridSkeleton key="product-grid-skeleton" slots={pageSize} />) : (<ul key="product-grid"> {products.map(...)}</ul>)
}
```